lib/HunkView.ml

```diff
@@ -105,53 +185,90 @@ let lines_with_numbers (lines : line list) (attr_change : Notty.attr)
-    (attr : Notty.attr) (change_type : [ `Add | `Remove ]) : Nottui.ui =
+    (attr : Notty.attr) (change_type : [ `Add | `Remove ]) : Ui.t =
   let sign = match change_type with `Add -> "+" | `Remove -> "-" in
-  if hunk_length > 0 then
-    W.string ~attr
-      (Printf.sprintf "@@ %s%d,%d @@" sign start_line_num hunk_length)
-  else W.string ~attr (Printf.sprintf "@@ %s0,0 @@" sign)
-
-let ui_of_hunk_side_by_side (hunk : string Patch.hunk) : Nottui.ui =
-  let attr_mine = Notty.A.(fg red ++ st bold) in
-  let attr_their = Notty.A.(fg green ++ st bold) in
-
-  let mine_lines, their_lines = split_and_align_hunk hunk.Patch.lines in
-
-  let content_mine = lines_with_numbers mine_lines attr_mine "-" in
-  let content_their = lines_with_numbers their_lines attr_their "+" in
-  let summary_mine =
-    create_summary
-      (hunk.Patch.mine_start + 1)
-      hunk.Patch.mine_len attr_mine `Remove
+  let summary =
+    Printf.sprintf "@@ %s%d,%d @@" sign start_line_num hunk_length
   in
-  let summary_their =
-    create_summary
-      (hunk.Patch.their_start + 1)
-      hunk.Patch.their_len attr_their `Add
+  W.string ~attr summary
+
+(* Main View Functions *)
+
+let ui_unified_diff (hunk : string Patch.hunk) (mode : rendering_mode) : Ui.t =
+  let hunk_summary = render_hunk_summary hunk mode in
+  let hunk_content =
+    let blocks = Block.of_hunk hunk.Patch.lines in
+    let single_line_changes =
+      List.for_all
+        (function
+          | Block.Changed { mine; their; _ } ->
+              List.length mine = 1 && List.length their = 1
+          | _ -> true)
+        blocks
+    in
+    if single_line_changes then
+      let word_diff_blocks = List.map WordDiff.compute blocks in
+      let word_diff_lines = Block.to_hunk word_diff_blocks in
+      render_hunk_lines word_diff_lines mode
+    else render_hunk hunk mode
   in
-  let space = Ui.space 1 0 in
-  Ui.hcat
-    [
-      Ui.resize ~w:0 ~sw:2 (Ui.vcat (summary_mine :: content_mine));
-      space;
-      Ui.resize ~w:0 ~sw:2 (Ui.vcat (summary_their :: content_their));
-    ]
+  Ui.vcat [ hunk_summary; hunk_content ]
 
-let current_hunks_side_by_side (z_patches : string Patch.t Zipper.t) : Nottui.ui
-    =
+let current_hunks (z_patches : string Patch.t Zipper.t) (mode : rendering_mode)
+    : Ui.t =
   let p = Zipper.get_focus z_patches in
-  let hunks_ui = List.map ui_of_hunk_side_by_side p.Patch.hunks in
-  Ui.vcat hunks_ui
+  let hunks = List.map (fun hunk -> ui_unified_diff hunk mode) p.Patch.hunks in
+  Ui.vcat hunks
+
+let current_hunks_side_by_side (z_patches : string Patch.t Zipper.t)
+    (mode : rendering_mode) : Ui.t =
+  let p = Zipper.get_focus z_patches in
+  let render_side_by_side hunk =
+    let mine_lines, their_lines = split_and_align_hunk hunk.Patch.lines in
+    let render_lines lines marker attr =
+      lines_with_numbers lines attr marker mode
+    in
+    let content_mine =
+      render_lines mine_lines removed_marker Notty.A.(fg red)
```

Comment:
`removed_marker` and `added_marker` are working with word in the context of word diff. It doesn't make sense to apply them to entire lines.